kv/memberlist/memberlist_client.go

```diff
@@ -193,6 +196,8 @@ func (cfg *KVConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.BoolVar(&cfg.EnableCompression, prefix+"memberlist.compression-enabled", mlDefaults.EnableCompression, "Enable message compression. This can be used to reduce bandwidth usage at the cost of slightly more CPU utilization.")
 	f.StringVar(&cfg.AdvertiseAddr, prefix+"memberlist.advertise-addr", mlDefaults.AdvertiseAddr, "Gossip address to advertise to other members in the cluster. Used for NAT traversal.")
 	f.IntVar(&cfg.AdvertisePort, prefix+"memberlist.advertise-port", mlDefaults.AdvertisePort, "Gossip port to advertise to other members in the cluster. Used for NAT traversal.")
+	f.StringVar(&cfg.Label, prefix+"memberlist.label", mlDefaults.Label, "Label is an optional set of bytes to include on the outside of each packet and stream.")
```

Comment:
The "label" CLI flag and YAML config option name is very generic. I'm wondering if we should call it something like "cluster-label" or "cluster-id" given we use it to authenticate the cluster ID. If so, then I would change the "skip" CLI/YAML option accordingly.

Reply:
`cluster-label` has a nice sound to it

I think the config option (which inverts the behavior to memberlist unfortunately) which I would prefer would be

`enforce-cluster-label`

Although we might want to get more opinions on `enforce` i'm not sure how well that would be understood for non native english speakers and if there is something more clear?

Reply:
(Also enforce-cluster-label would default to true IMO)

Reply:
I like "cluster-label" too. While working on the jsonnet changes to configure it, it sounds nice in the config.

Reply:
> enforce-cluster-label


WDYT about "cluster-label-verification-enabled" instead of "enforce-cluster-label"? While doing change I've the feeling "enforce-cluster-label" may not be that clear

Reply:
From Ed on Slack:
> sounds good to me! I like that better

Reply:
From Ed on Slack:

> also you could do it then as `cluster-label-verification-disabled`
> and the default could be false and it would then match the same semantics as the existing config in memberlist